File: dart-wrapper/cardinal_sdk/lib/auth/authentication_method.dart
import 'package:cardinal_sdk/auth/credentials.dart';

sealed class AuthenticationMethod {

  static Map<String, dynamic> encode(AuthenticationMethod value) {
    switch (value) {
      case UsingCredentials v:
        final encodedValue = UsingCredentials.encode(v);
        encodedValue["type"] = "com.icure.cardinal.sdk.dart.options.AuthenticationMethod.UsingCredentials";
        return encodedValue;
    }
  }

}

class UsingCredentials extends AuthenticationMethod {
  final Credentials _credentials;

  UsingCredentials(this._credentials);

  static Map<String, dynamic> encode(UsingCredentials value) {
    return {
      "credentials": Credentials.encode(value._credentials)
    };
  }
}

File: dart-wrapper/lib/src/commonMain/kotlin/com/icure/cardinal/sdk/dart/auth/AuthenticationMethod.kt
package com.icure.cardinal.sdk.dart.auth

import kotlinx.serialization.Serializable

@Serializable
sealed interface AuthenticationMethod {

	fun toMultiplatform(): com.icure.cardinal.sdk.options.AuthenticationMethod

	@Serializable
	data class UsingCredentials(val credentials: Credentials) : AuthenticationMethod {
		override fun toMultiplatform() =
			com.icure.cardinal.sdk.options.AuthenticationMethod.UsingCredentials(credentials.toKt())
	}

}